tests/unit/test_ut_scenario_loader.py

```diff
@@ -160,70 +160,94 @@ def test_is_def_filter(self):
+                MyTests().test_1_2_3_4_myscenario4alt()  # noqa pylint: disable=E1101
+                MyTests().test_1_2_3_4_5_myscenario5()  # noqa pylint: disable=E1101
+                MyTests().test_1_2_3_4_5_myscenario5alt()  # noqa pylint: disable=E1101
+
+                raised = False
+                try:
+                    MyTests().test_1_2_3_4_5_myscenario100()
+                except AttributeError:
+                    raised = True
+
+                self.assertTrue(raised)
             finally:
                 HotSOSConfig.set(**orig_config)
 
     def test_scenarios_deep(self):
-        with tempfile.TemporaryDirectory() as dtmp:
+        with tempfile.TemporaryDirectory() as dtmp,\
+             mock.patch.object(utils, 'DEFS_DIR', dtmp),\
+             mock.patch.object(utils, 'DEFS_TESTS_DIR', dtmp + '/tests'):
             self.create_scenarios(dtmp, levels=5)
             self.create_tests(dtmp, levels=5)
             orig_config = HotSOSConfig.CONFIG
             try:
                 HotSOSConfig.plugin_yaml_defs = dtmp
                 HotSOSConfig.global_tmp_dir = dtmp
                 HotSOSConfig.plugin_tmp_dir = dtmp
-                with mock.patch.object(utils, 'DEFS_TESTS_DIR',
-                                       os.path.join(dtmp, 'tests')):
-                    @utils.load_templated_tests('scenarios/testplugin/'
-                                                '1/2/3/4/5')
-                    class MyTests(ScenarioTestsBase):
-                        pass
 
-                    MyTests().test_myscenario5()  # pylint: disable=E1101
-                    MyTests().test_myscenario5alt()  # pylint: disable=E1101
+                @utils.load_templated_tests('scenarios/testplugin/'
+                                            '1/2/3/4/5')
+                class MyTests(ScenarioTestsBase):
+                    pass
+
+                MyTests().test_myscenario5()  # pylint: disable=E1101
+                MyTests().test_myscenario5alt()  # pylint: disable=E1101
 
-                    raised = False
-                    try:
-                        MyTests().test_1_myscenario1()
-                    except AttributeError:
-                        raised = True
+                raised = False
+                try:
+                    MyTests().test_1_myscenario1()
+                except AttributeError:
+                    raised = True
 
-                    self.assertTrue(raised)
+                self.assertTrue(raised)
             finally:
                 HotSOSConfig.set(**orig_config)
+
+    def test_scenarios_no_such_target(self):
+        with tempfile.TemporaryDirectory() as dtmp,\
+             mock.patch.object(utils, 'DEFS_DIR', dtmp),\
+             mock.patch.object(utils, 'DEFS_TESTS_DIR', dtmp + '/tests'):
+            self.create_tests(dtmp, levels=5)
+            # Do not create scenarios intentionally
+            try:
```

Comment:
You can use ```with self.assertRaises(FileNotFoundError):``` and I also realise I should have done this in the above tests.

Reply:
Will do